src/client/datascience/variablesView/variableView.ts

```diff
@@ -78,29 +113,62 @@ export class VariableView extends WebviewViewHost<IVariableViewPanelMapping> imp
         handler.bind(this)(args);
     }
 
-    // This is called when the UI side requests new variable data
+    // Handle a request from the react UI to show our data viewer
+    private async showDataViewer(request: IShowDataViewer): Promise<void> {
+        try {
+            if (
+                this.notebookEditorProvider.activeEditor &&
+                this.notebookEditorProvider.activeEditor.notebook &&
+                (await this.dataViewerChecker.isRequestedColumnSizeAllowed(request.columnSize, this.owningResource))
+            ) {
+                // Create a variable data provider and pass it to the data viewer factory to create the data viewer
+                const jupyterVariableDataProvider = await this.jupyterVariableDataProviderFactory.create(
+                    request.variable,
+                    this.notebookEditorProvider.activeEditor.notebook
+                );
+                const title: string = `${localize.DataScience.dataExplorerTitle()} - ${request.variable.name}`;
+                await this.dataViewerFactory.create(jupyterVariableDataProvider, title);
+            }
+        } catch (e) {
+            traceError(e);
+            this.appShell.showErrorMessage(e.toString());
```

Comment:
I'd prefer to have a better error message than just a stringified error.
E.g. `Failed to display data viewer, please check logs for further info...`.

Should we optionally capture telemetry here.
Else we're expecting users to file bugs if somethign goes wrong.
With the telemtry we'd know how oftent this is failing for users.

Reply:
Yeah, I'd just pulled this from the old code, but no time like now to fix it up. I'll change the message here and in the other location and add a telemetry point.